packages/core/src/api/blockManipulation/tables/tables.ts

```
@@ -224,78 +388,344 @@ export function resolveAbsoluteTableCellIndices(
-    block
-  );
+  try {
+    const occupancyGrid = getTableCellOccupancyGrid(block);
 
-  // Then for each column, get the cell at the absolute row index as a relative cell index
-  const cells = new Array(width).fill(false).map((_v, col) => {
-    return resolveAbsoluteTableCellIndices(
-      { row: absoluteRow, col },
-      block,
-      occupancyGrid
-    );
-  });
-  // Filter out duplicates based on row and col properties
-  return cells.filter((cell, index): cell is NonNullable<typeof cell> => {
-    return (
-      cell !== null &&
-      cells.findIndex((c) => c && c.row === cell.row && c.col === cell.col) ===
+    // First need to resolve the relative row index to an absolute row index
+    let absoluteRow = 0;
+
+    // Jump through the occupied cells ${relativeCellIndices.row} times to find the absolute row position
+    for (let i = 0; i < relativeRowIndex; i++) {
+      const cell = occupancyGrid[absoluteRow]?.[0];
+
+      if (!cell) {
+        // As a sanity check, if the cell is not occupied, we should throw an error
+        throw new Error(
+          `Unable to resolve relative table cell indices for table, cell at ${absoluteRow},0 is not occupied`
+        );
+      }
+
+      // Skip the cells that the rowspan takes up
+      absoluteRow += cell.rowspan;
+    }
+
+    // Then for each column, get the cell at the absolute row index as a relative cell index
+    const cells = new Array(occupancyGrid[0].length)
+      .fill(false)
+      .map((_v, col) => {
+        return getRelativeTableCellIndices(
+          { row: absoluteRow, col },
+          block,
+          occupancyGrid
+        );
+      });
+
+    // Filter out duplicates based on row and col properties
+    return cells.filter((cell, index) => {
+      return (
+        cells.findIndex((c) => c.row === cell.row && c.col === cell.col) ===
         index
-    );
-  });
+      );
+    });
+  } catch (e) {
+    // In case of an invalid index, return an empty array
+    return [];
```

can this happen? otherwise better to just let the error bubble